src/imagecapturecore.cs

```diff
@@ -0,0 +1,900 @@
+
+		[Field ("ICSavedFilename")]
+		NSString SavedFilename { get; }
+
+		[Field ("ICSavedAncillaryFiles")]
+		NSString SavedAncillaryFiles { get; }
+
+		[Field ("ICOverwrite")]
+		NSString Overwrite { get; }
+
+		[Field ("ICDeleteAfterSuccessfulDownload")]
+		NSString DeleteAfterSuccessfulDownload { get; }
+
+		[Field ("ICDownloadSidecarFiles")]
+		NSString DownloadSidecarFiles { get; }
+	}
+
+	[Static]
+	interface ICStatusNotificationKeys { // Keys into DidReceiveStatusInformation dictionary
+		[Field ("ICStatusNotificationKey")]
+		NSString NotificationKey { get; }
+
+		[Field ("ICStatusCodeKey")]
+		NSString CodeKey { get; }
+
+		[Field ("ICLocalizedStatusNotificationKey")]
+		NSString LocalizedNotificationKey { get; }
+	}
+
+	[Static]
+	interface ICButtonType {
+		[Field ("ICButtonTypeScan")]
+		NSString Scan { get; }
+
+		[Field ("ICButtonTypeMail")]
+		NSString Mail { get; }
+
+		[Field ("ICButtonTypeCopy")]
+		NSString Copy { get; }
+
+		[Field ("ICButtonTypeWeb")]
+		NSString Web { get; }
+
+		[Field ("ICButtonTypePrint")]
+		NSString Print { get; }
+
+		[Field ("ICButtonTypeTransfer")]
+		NSString Transfer { get; }
+	}
+
+	[Protocol, Model (AutoGeneratedName = true)]
+	[BaseType (typeof(NSObject))]
+	interface ICDeviceDelegate {
+
+		[Abstract]
+		[Export ("didRemoveDevice:")]
+		void DidRemoveDevice (ICDevice device);
+
+		[Export ("device:didOpenSessionWithError:")]
+		void DidOpenSession (ICDevice device, [NullAllowed] NSError error);
+
+		[Export ("deviceDidBecomeReady:")]
+		void DidBecomeReady (ICDevice device);
+
+		[Export ("device:didCloseSessionWithError:")]
+		void DidCloseSession (ICDevice device, [NullAllowed] NSError error);
+
+		[Export ("deviceDidChangeName:")]
+		void DidChangeName (ICDevice device);
+
+		[Deprecated (PlatformName.MacOSX, 10, 13)]
+		[Export ("deviceDidChangeSharingState:")]
+		void DidChangeSharingState (ICDevice device);
+
+		[Export ("device:didReceiveStatusInformation:")]
+		void DidReceiveStatusInformation (ICDevice device, NSDictionary<NSString, NSObject> status);
+
+		[Export ("device:didEncounterError:")]
+		void DidEncounterError (ICDevice device, [NullAllowed] NSError error);
+
+		[Export ("device:didReceiveButtonPress:")]
+		void DidReceiveButtonPress (ICDevice device, NSString /* ICButtonType */ buttonType);
+
+		[Export ("device:didReceiveCustomNotification:data:")]
+		void DidReceiveCustomNotification (ICDevice device, NSDictionary<NSString, NSObject> notification, NSData data);
+	}
+
+	[BaseType (typeof(NSObject))]
+	interface ICDevice {
+
+		[Wrap ("WeakDelegate")]
+		[NullAllowed]
+		IICDeviceDelegate Delegate { get; set; }
+
+		[NullAllowed, Export ("delegate", ArgumentSemantic.Assign)]
+		NSObject WeakDelegate { get; set; }
+
+		[Export ("type")]
+		ICDeviceType Type { get; }
+
+		[NullAllowed, Export ("name")]
+		string Name { get; }
+
+		[NullAllowed, Export ("icon")]
+		CGImage Icon { get; }
+
+		[Export ("capabilities")]
+		string[] Capabilities { get; }
```

Comment:
if that's related to `ICDeviceCapabilities` then it should be exposed as `NSString[]` otherwise if will force conversions between the types.